dbt/adapters/bigquery/impl.py

```diff
@@ -564,18 +590,15 @@ def _partitions_match(table, conf_partition: Optional[PartitionConfig]) -> bool:
         if not is_partitioned and not conf_partition:
             return True
         elif conf_partition and table.time_partitioning is not None:
-            partitioning_field = table.time_partitioning.field or "_PARTITIONTIME"
-            table_field = partitioning_field.lower()
-            table_granularity = table.partitioning_type.lower()
-            conf_table_field = (
-                conf_partition.field
-                if not conf_partition.time_ingestion_partitioning
-                else "_PARTITIONTIME"
+            table_field = (
```

Comment:
I agree that it can be confusing to understand if it's meant to be a class or str, maybe being more explicit on the naming (eg `table_field_name`).
Also some overall field naming could be improved.